datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -177,35 +177,12 @@ pub fn can_coerce_from(type_into: &DataType, type_from: &DataType) -> bool {
         Timestamp(TimeUnit::Nanosecond, None) => {
             matches!(type_from, Null | Timestamp(_, None))
         }
-        Date32 => {
```

Comment:
why this change?

Though I admit that the fact that all tests still pass is concerning 🤔

Reply:
This was added from the commit https://github.com/comphead/arrow-datafusion/blob/4761802a4bc46d377406aa20767c4af66b80d65b/datafusion/expr/src/type_coercion/functions.rs#L180 in the https://github.com/apache/arrow-datafusion/pull/4726

I just revert useless codes adding from the PR https://github.com/apache/arrow-datafusion/pull/4726